### PR TITLE
deps: Update S3 to version 2.25.30

### PR DESCRIPTION
**Pull request by MarcelBochtler**
This updated version updates netty-codec-http to version `4.1.108.Final` which fixed the vulnerability CVE-2024-29025 [1].
Unfortunately, no release notes are published.

[1]: https://nvd.nist.gov/vuln/detail/CVE-2024-29025

Please ensure that your pull request adheres to our [contribution guidelines](https://github.com/oss-review-toolkit/.github/blob/main/CONTRIBUTING.md). Thank you!

### CODE REVIEW

**Review by tsteenbe — CHANGES_REQUESTED** [review]
Was a bit confused when I first read this PR so believe commit message can be improved. @MarcelBochtler  What do you think of the following?

```
deps: Update S3 to version 2.25.28

This newer version uses netty-codec-http version `4.1.108.Final`
which includes a fix for the CVE-2024-29025 known vulnerability [1] .

Unfortunately, no release notes are published.

[1]: https://github.com/advisories/GHSA-5jpm-x58v-624v
```

**Comment by sschuberth** [comment]
> Unfortunately, no release notes are published.

Aren't [these](https://github.com/aws/aws-sdk-java-v2/releases/tag/2.25.28) the release notes? They don't mention the netty-codec-http update, though.

**Comment by sschuberth** [comment]
And actually, [this](https://github.com/aws/aws-sdk-java-v2/commit/3dcb0ec89e4a598328ed8438207e7cff1e468a31) looks like Netty was already updated back in version [2.25.19](https://github.com/aws/aws-sdk-java-v2/releases/tag/2.25.19). So if your argument is only to remedy the CVE, then updating to that version would be sufficient. At least you should not make it sound as if version 2.25.28 would be required to address the issue.

**Comment by MarcelBochtler** [comment]
> > Unfortunately, no release notes are published.
> 
> Aren't [these](https://github.com/aws/aws-sdk-java-v2/releases/tag/2.25.28) the release notes? They don't mention the netty-codec-http update, though.

Thanks, I was somehow not able to find them.

> And actually, [this](https://github.com/aws/aws-sdk-java-v2/commit/3dcb0ec89e4a598328ed8438207e7cff1e468a31) looks like Netty was already updated back in version [2.25.19](https://github.com/aws/aws-sdk-java-v2/releases/tag/2.25.19). So if your argument is only to remedy the CVE, then updating to that version would be sufficient. At least you should not make it sound as if version 2.25.28 would be required to address the issue.

My reason to upgrade was the CVE.
And when upgrading I always use the latest version.

I updated the commit message, trying to clarify this.

> Was a bit confused when I first read this PR so believe commit message can be improved. @MarcelBochtler What do you think of the following?

Yes, the previous commit message was a bit confusing. I updated it.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -51,11 +51,12 @@ postgres = "42.7.3"
...
 reflections = "0.10.2"
 retrofit = "2.11.0"
-s3 = "2.25.0"
+s3 = "2.25.30"
 saxonHe = "12.4"
 scanoss = "1.1.6"
 semver4j = "5.2.3"
 slf4j = "2.0.12"
+
 springCore = "5.3.33"
```
Comment (fviernau) [inline gradle/libs.versions.toml:60]: Unrelated change in line 59.

**Review by fviernau — CHANGES_REQUESTED** [review]

**Pull request merged**